src/libguac/wol.c

```diff
@@ -83,35 +84,68 @@ static ssize_t __guac_wol_send_packet(const char* broadcast_addr,
     /* Determine the IP version, starting with IPv4. */
     wol_dest.sin_port = htons(GUAC_WOL_PORT);
     wol_dest.sin_family = AF_INET;
-    if (inet_pton(AF_INET, broadcast_addr, &(wol_dest.sin_addr)) == 0) {
+    int retval = inet_pton(wol_dest.sin_family, broadcast_addr, &(wol_dest.sin_addr));
+    
+    /* If return value is less than zero, the address doesn't match any known family. */
+    if (retval < 0) {
+        guac_error = GUAC_STATUS_SEE_ERRNO;
+        guac_error_message = "Unknown broadcast or multicast address type specified for Wake-on-LAN";
+        return 0;
+    }
+    
+    /* If return value is zero, address doesn't match the IPv4, so try IPv6. */
+    else if (retval == 0) {
         wol_dest.sin_family = AF_INET6;
-        if (inet_pton(AF_INET6, broadcast_addr, &(wol_dest.sin_addr)) == 0)
+        retval = inet_pton(wol_dest.sin_family, broadcast_addr, &(wol_dest.sin_addr));
+        
+        /* IPv6 didn't work, either, so bail out. */
+        if (retval == 0) {
```

Comment:
Does `retval < 0` need to be checked for this call to `inet_pton()`, as well?

Reply:
Ah, I was mis-reading what the man page said about the `-1` return value of `inet_pton()`.  You're right, let me fix that real quick...